packages/flutter/test/material/expansion_tile_test.dart

```diff
@@ -229,4 +229,69 @@ void main() {
 
     expect(find.text('Subtitle'), findsOneWidget);
   });
+
+  testWidgets('ExpansionTile custom tween colors', (WidgetTester tester) async {
+    final Key expandedTitleKey = UniqueKey();
+    final Key collapsedTitleKey = UniqueKey();
+    final Key expandedIconKey = UniqueKey();
+    final Key collapsedIconKey = UniqueKey();
+    const Color expandedTitleColor = Colors.blue;
+    const Color collapsedTitleColor = Colors.red;
+    const Color expandedIconColor = Colors.green;
+    const Color collapsedIconColor = Colors.orange;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: SingleChildScrollView(
+            child: Column(
+              children: <Widget>[
+                const ListTile(title: Text('Top')),
+                ExpansionTile(
+                  initiallyExpanded: true,
+                  headerColorTweenBegin: collapsedTitleColor,
+                  headerColorTweenEnd: expandedTitleColor,
+                  iconColorTweenBegin: collapsedIconColor,
+                  iconColorTweenEnd: expandedIconColor,
+                  title: TestText('Expanded', key: expandedTitleKey),
+                  children: const <Widget>[ListTile(title: Text('0'))],
+                  trailing: TestIcon(key: expandedIconKey),
+                ),
+                ExpansionTile(
+                  initiallyExpanded: false,
+                  headerColorTweenBegin: collapsedTitleColor,
+                  headerColorTweenEnd: expandedTitleColor,
+                  iconColorTweenBegin: collapsedIconColor,
+                  iconColorTweenEnd: expandedIconColor,
+                  title: TestText('Collapsed', key: collapsedTitleKey),
+                  children: const <Widget>[ListTile(title: Text('0'))],
+                  trailing: TestIcon(key: collapsedIconKey),
+                ),
+              ],
+            ),
+          ),
+        ),
+      ),
+    );
+
+    Color iconColor(Key key) => tester.state<TestIconState>(find.byKey(key)).iconTheme.color;
+    Color textColor(Key key) => tester.state<TestTextState>(find.byKey(key)).textStyle.color;
+
+    expect(textColor(expandedTitleKey), expandedTitleColor);
+    expect(textColor(collapsedTitleKey), collapsedTitleColor);
+    expect(iconColor(expandedIconKey), expandedIconColor);
+    expect(iconColor(collapsedIconKey), collapsedIconColor);
+
+    // Tap both tiles to change their state: collapse and expand respectively
+    await tester.tap(find.text('Expanded'));
+    await tester.tap(find.text('Collapsed'));
+    await tester.pump();
+    await tester.pump(const Duration(seconds: 1));
+    await tester.pump(const Duration(seconds: 1));
+
```

Comment:
We should probably test at least one color value mid-transition as well to ensure that the the color is properly lerp-ing from start to finish. A quick example I could find is [here](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/theme_test.dart#L289). Check and see how long the transition takes and jump halfway through it, making sure the color is the expected value.

Reply:
Do you still think this is valid or are the current tests enough?

Reply:
Yeah, I still think it's important since there isn't any part of the test that checks for the animating colors. It should be pretty straightforward to implement, but let me know if you run into any issues!